back/src/domains/establishment/use-cases/AssessmentReminder.ts

```diff
@@ -0,0 +1,197 @@
+import subDays from "date-fns/subDays";
+import { difference } from "ramda";
+import {
+  AbsoluteUrl,
+  AgencyRole,
+  ConventionId,
+  ConventionReadDto,
+  Email,
+  errors,
+  executeInSequence,
+  frontRoutes,
+  immersionFacileNoReplyEmailSender,
+} from "shared";
+import { z } from "zod";
+import { GenerateConventionMagicLinkUrl } from "../../../config/bootstrap/magicLinkUrl";
+import { AssessmentRepository } from "../../convention/ports/AssessmentRepository";
+import { ConventionRepository } from "../../convention/ports/ConventionRepository";
+import { createTransactionalUseCase } from "../../core/UseCase";
+import {
+  NotificationContentAndFollowedIds,
+  SaveNotificationAndRelatedEvent,
+} from "../../core/notifications/helpers/Notification";
+import { TimeGateway } from "../../core/time-gateway/ports/TimeGateway";
+import { UnitOfWork } from "../../core/unit-of-work/ports/UnitOfWork";
+
+type AssessmentReminderOutput = {
+  numberOfReminders: number;
+};
+
+export type AssessmentReminderMode =
+  | "3daysAfterConventionEnd"
+  | "10daysAfterConventionEnd";
+
+export type AssessmentReminder = ReturnType<typeof makeAssessmentReminder>;
+export const makeAssessmentReminder = createTransactionalUseCase<
+  { mode: AssessmentReminderMode },
+  AssessmentReminderOutput,
+  void,
+  {
+    timeGateway: TimeGateway;
+    saveNotificationAndRelatedEvent: SaveNotificationAndRelatedEvent;
+    generateConventionMagicLinkUrl: GenerateConventionMagicLinkUrl;
+  }
+>(
+  {
+    name: "AssessmentReminder",
+    inputSchema: z.object({
+      mode: z.enum(["3daysAfterConventionEnd", "10daysAfterConventionEnd"]),
+    }),
+  },
+  async ({ inputParams: params, uow, deps }) => {
+    const now = deps.timeGateway.now();
+    const conventionIdsToRemind = await getConventionIdsToRemind({
+      mode: params.mode,
+      now,
+      assessmentRepository: uow.assessmentRepository,
+      conventionRepository: uow.conventionRepository,
+    });
+
+    await Promise.all([
+      executeInSequence(conventionIdsToRemind, async (conventionId) => {
```

Comment:
il me semble que execute in sequence renvoie une seule promise, donc tu peux juste faire ça :

```suggestion
    await executeInSequence(conventionIdsToRemind, async (conventionId) => {
```